backend/BalanceBuddy/src/main/java/com/example/balancebuddy/services/UserService.java

```diff
@@ -39,4 +50,80 @@ public Optional<MyUser> findByEmail(String email) {
         return result.isEmpty() ? Optional.empty() : Optional.of(result.get(0));
     }
 
+    @Transactional
+    public void deleteUserByEmail(String email) {
+        findByEmail(email).ifPresent(entityManager::remove);
+    }
+
+    @Transactional
+    public MyUser updateUser(MyUser user) {
+        MyUser existingUser = entityManager.find(MyUser.class, user.getUserID());
+
+        if (existingUser != null) {
+            existingUser.setFirstname(user.getFirstname());
+            existingUser.setLastname(user.getLastname());
+            existingUser.setEmail(user.getEmail());
+            existingUser.setPassword(passwordEncoder.encode(user.getPassword()));
+            entityManager.merge(existingUser);
+            return existingUser;
+        } else {
+            throw new IllegalArgumentException("User not found!");
+        }
+    }
+
+    @Transactional
+    public void myChangePassword(String email, String oldPassword, String newPassword) throws Exception {
+        Optional<MyUser> userOptional = findByEmail(email);
+
+        // If the user exists in the database, continue changing the passwords
+        if (userOptional.isPresent()) {
+            MyUser user = userOptional.get();
+            if (passwordEncoder.matches(oldPassword, user.getPassword())) {
+                if (oldPassword.equals(newPassword)) {
+                    throw new Exception("New password cannot be the same as the old password");
+                }
+                user.setPassword(passwordEncoder.encode(newPassword));
+                entityManager.merge(user);
+            } else {
+                throw new IllegalArgumentException("Old password is incorrect!");
+            }
+        } else {
+            throw new UsernameNotFoundException("User not found!");
+        }
+    }
+
+    public boolean userExists(String email) {
+        return findByEmail(email).isPresent();
+    }
+
+    public MyUser loadUserByUsername(String email) throws UsernameNotFoundException {
+        Optional<MyUser> userOptional = findByEmail(email);
+
+        if (userOptional.isEmpty()) {
+            throw new UsernameNotFoundException("User with email " + email + " not found");
+        }
+        return userOptional.get();
+    }
+
+    @Override
+    public void createUser(UserDetails userDetails) {
+        MyUser myUser = (MyUser) userDetails;
+        createUser(myUser);
+    }
+
+    @Override
+    public void updateUser(UserDetails userDetails) {
+        MyUser myUser = (MyUser) userDetails;
+        updateUser(myUser);
+    }
+
+    @Override
+    public void deleteUser(String username) {
+        deleteUserByEmail(username);
+    }

```

Comment:
It seems that you have another createUser and updateUser method (that use MyUser object) which are public. If you don't intend to use those 2 methods outside this class, I would recommend you to do the following:

- You can add the logic from the methods where you return MyUser object in the methods where you return void and add @Transactional annotation above @Override.
- You can do the same for deleting the user by email as well.

This way you won't have overloaded methods anymore and you will ensure better readability of your code.